src/plugins/WebOfTrust/util/BackgroundJobFactory.java

```diff
@@ -0,0 +1,50 @@
+package plugins.WebOfTrust.util;
+
+/**
+ * Factory for {@link BackgroundJob}s which allows all background jobs constructed using an instance
+ * of this class to be terminated with a single method invocation.
+ *
+ * @author bertm
+ * @see BackgroundJobFactoryBase
+ */
+public interface BackgroundJobFactory {
+    /**
+     * Constructs a new background job using the default parameters. When this background job is
+     * {@link BackgroundJob#terminate() terminated}, the running job will be notified by
+     * interruption of its thread. Hence, the job implementer must take care not to swallow
+     * {@link InterruptedException}.
```

Comment:
The latter suggestion is too specific IMHO. We can't dictate what background job (and its termination) will be used for.  I'll rephrase as something like "don't swallow, check interrupted flag and exit accordingly".